fix: false positive errors when validating examples with refs (#1377)
Redocly/redocly-cli: .changeset/small-rules-vanish.md

---
"@redocly/openapi-core": patch
"@redocly/cli": patch
---

> Fixed false positive errors when validating media type examples that contain references.

Reviewer: So that I understand this properly: we fixed a problem where no errors were reported - but the examples were wrong? So users might see errors without changing their API description, is that correct? Was it specifically when the changes went across files?

Author: >  So users might see errors without changing their API description, is that correct?

Actually vice versa. We show an error when there's no error (due to wrong references resolving). After the update, false errors should be gone.

Reviewer: OK let's update the changelog entry to describe what the user might experience.

> Fixed false positive errors when validating media type examples that contain references.

Author: ```suggestion
Fixed false positive errors when validating media type examples. Now, the linter won't error on valid examples that contain references. 
```

> Fixed false positive errors when validating media type examples. Now, the linter won't error on valid examples that contain references.

Reviewer 2: ```suggestion
Fixed false positive errors with valid media type examples that contain references.
```
I think we should avoid "Now," in the changelogs and docs in general. Now, what do you think about that? 😆

Author: Let's have Roman's review on the solution firstly. It might turn out we don't need any changeset at all 🤣

Reviewer 3: I checked it. LGTM.

Reviewer: I'm sorry, I did already edit this changeset once, and while I agree that it's better without the "Now", I think it lost some context. Combining the two versions, how about this?
```suggestion
Fixed a problem where the linter incorrectly returned an error for valid examples that contain references.
```

Reviewer 3: I'm sure pretty sure he is good to 😄

Go ahead and accept the suggestion + merge.
